Token exchange docs (#602)
Shopify/shopify-app-js: packages/shopify-app-remix/README.md

You can also authenticate requests from surfaces other than the admin.
To see all supported methods, see [the `shopify.authenticate` object documentation](https://shopify.dev/docs/api/shopify-app-remix/latest/authenticate).

> ### New embedded authorization strategy

Reviewer: Nit: `New embedded app authorization strategy`

It replaces the existing [installation and authorization code grant flow](https://shopify.dev/docs/apps/auth/get-access-tokens/authorization-code-grant).

This is achieved by using [Shopify managed installation](https://shopify.dev/docs/apps/auth/installation#shopify-managed-installation)
to handle automatic app installations and scope updates, while utilizing
[token exchange](https://shopify.dev/docs/apps/auth/get-access-tokens/token-exchange/overview) to retrieve an access token for
authenticated API access.

##### Enabling this new strategy in your app
> [!NOTE]
> Newly created Remix apps from the template after February 1st 2024 will have this feature enabled by default.

1. Enable [Shopify managed installation](https://shopify.dev/docs/apps/auth/installation#shopify-managed-installation)
by configuring your scopes [through the Shopify CLI](https://shopify.dev/docs/apps/tools/cli/configuration).
2. Enable the future flag `unstable_newEmbeddedAuthStrategy` in your app's server configuration file.

```ts
// my-app/app/shopify.server.ts
const shopify = shopifyApp({
  ...
  isEmbeddedApp: true,
  future: {
    unstable_newEmbeddedAuthStrategy: true,
  }
})

```

> 3. Enjoy no-redirect OAuth flow, and app installation process.

Reviewer: ```suggestion
3. Enjoy a smoother and faster app installation process.
```
